app/Dockerfile

```diff
@@ -46,8 +46,14 @@ RUN poetry check
 COPY ./package.json /app/package.json
 COPY ./yarn.lock /app/yarn.lock
 COPY ./experimenter/legacy-ui/core/package.json /app/experimenter/legacy-ui/core/package.json
+RUN yarn install --frozen-lockfile
+
 COPY ./experimenter/legacy-ui/rapid/package.json /app/experimenter/legacy-ui/rapid/package.json
+RUN yarn install --frozen-lockfile
+
 COPY ./experimenter/legacy-ui/visualization/package.json /app/experimenter/legacy-ui/visualization/package.json
+RUN yarn install --frozen-lockfile
```

Comment:
Will a `yarn install` at the app root not work (over all the workspaces, or does that not exist in this context)?

Reply:
Yeah so, the way it was setup was copy all the package.jsons in and then install all packages, but if you change something in just nimbus-ui/package.json it'll invalidate that whole cache layer and reinstall everything, so i broke it up to cache each of those layers separately, so if we make any changes to just the nimbus one it won't invalidate the others, which should make things a little quick for local dev, and also speed up dependabot prs.  Just a small performance improvement.